src/mono/wasm/Wasm.Build.Tests/NativeBuildTests.cs

```diff
@@ -133,5 +135,50 @@ public void BlazorWasm_CanRunMonoAOTCross_WithNoTrimming(string config)
             Assert.True(res.ExitCode != 0, "Expected publish to fail");
             Assert.Contains("Stopping after AOT", res.Output);
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void BuildWithUndefinedNativeSymbol(bool allowUndefined)
+        {
+            string id = $"UndefinedNativeSymbol_{(allowUndefined ? "allowed" : "disabled")}_{Path.GetRandomFileName()}";
+
+            string code = @"
+                using System;
+                using System.Runtime.InteropServices;
+
+                call();
+                return 42;
+
+                [DllImport(""undefined"")] static extern void call();
+            ";
+
+            CreateWasmTemplateProject(id);
+
+            AddItemsPropertiesToProject(
+                Path.Combine(_projectDir!, $"{id}.csproj"),
+                extraItems: @$"<NativeFileReference Include=""undefined.c"" />",
+                extraProperties: null
+            );
+
+            File.WriteAllText(Path.Combine(_projectDir!, "Program.cs"), code);
+            File.Copy(Path.Combine(BuildEnvironment.TestAssetsPath, "native-libs", "undefined-symbol.c"), Path.Combine(_projectDir!, "undefined.c"));
+
+            CommandResult result = new DotNetCommand(s_buildEnv, _testOutput)
+                .WithWorkingDirectory(_projectDir!)
+                .WithEnvironmentVariable("NUGET_PACKAGES", _nugetPackagesDir)
+                .ExecuteWithCapturedOutput("build", "-c Release", allowUndefined ? $"-p:WasmAllowUndefinedSymbols=true" : string.Empty);
```

Comment:
Set this property in the project file itself. The difference is that when you pass it on the command line, the value is forced to remain the same, and if the targets are trying to change the value then msbuild will ignore that. IOW, we won't be testing any defaults or overrides we might have in our targets.